src/axom/klee/Geometry.cpp

```diff
@@ -6,29 +6,136 @@
+  , m_format("blueprint-tets")
+  , m_path()
+  , m_meshGroup(meshGroup)
+  , m_topology(topology)
+  , m_levelOfRefinement(0)
+  , m_operator(std::move(operator_))
+{
+#ifdef AXOM_DEBUG
+  #if 0
+  SLIC_ASSERT_MSG(isBlueprintTetMesh(m_meshGroup),
+                  "Mesh provided to Geometry is not a valid blueprint "
+                  "unstructured tetrahedral mesh.");
+  #endif
+#endif
+}
+
+Geometry::Geometry(const TransformableGeometryProperties& startProperties,
+                   const axom::primal::Tetrahedron<double, 3>& tet,
+                   std::shared_ptr<GeometryOperator const> operator_)
+  : m_startProperties(startProperties)
+  , m_format("tet3D")
+  , m_path()
+  , m_meshGroup(nullptr)
+  , m_topology()
+  , m_tet(tet)
+  , m_levelOfRefinement(0)
+  , m_operator(std::move(operator_))
+{ }
+
+Geometry::Geometry(const TransformableGeometryProperties& startProperties,
+                   const axom::primal::Hexahedron<double, 3>& hex,
+                   std::shared_ptr<GeometryOperator const> operator_)
+  : m_startProperties(startProperties)
+  , m_format("hex3D")
+  , m_path()
+  , m_meshGroup(nullptr)
+  , m_topology()
+  , m_hex(hex)
+  , m_levelOfRefinement(0)
+  , m_operator(std::move(operator_))
+{ }
+
+Geometry::Geometry(const TransformableGeometryProperties& startProperties,
+                   const Sphere3D& sphere,
+                   axom::IndexType levelOfRefinement,
+                   std::shared_ptr<GeometryOperator const> operator_)
+  : m_startProperties(startProperties)
+  , m_format("sphere3D")
+  , m_path()
+  , m_meshGroup(nullptr)
+  , m_topology()
+  , m_sphere(sphere)
+  , m_levelOfRefinement(levelOfRefinement)
+  , m_operator(std::move(operator_))
+{ }
+
+Geometry::Geometry(const TransformableGeometryProperties& startProperties,
+                   const axom::Array<double, 2>& discreteFunction,
+                   const Point3D& vorBase,
+                   const Vector3D& vorDirection,
+                   axom::IndexType levelOfRefinement,
+                   std::shared_ptr<GeometryOperator const> operator_)
+  : m_startProperties(startProperties)
+  , m_format("vor3D")
+  , m_path()
+  , m_meshGroup(nullptr)
+  , m_topology()
+  , m_sphere()
+  , m_discreteFunction(discreteFunction)
+  , m_vorBase(vorBase)
+  , m_vorDirection(vorDirection)
+  , m_levelOfRefinement(levelOfRefinement)
+  , m_operator(std::move(operator_))
+{ }
+
+Geometry::Geometry(const TransformableGeometryProperties& startProperties,
+                   const axom::primal::Plane<double, 3>& plane,
+                   std::shared_ptr<GeometryOperator const> operator_)
+  : m_startProperties(startProperties)
+  , m_format("plane3D")
+  , m_path()
+  , m_meshGroup(nullptr)
+  , m_topology()
+  , m_plane(plane)
+  , m_levelOfRefinement(0)
+  , m_operator(std::move(operator_))
+{ }
+
+bool Geometry::hasGeometry() const
+{
+  bool isInMemory = m_format == "blueprint-tets" || m_format == "sphere3D" ||
```

Comment:
How often is this method called? Could the large number of string compares and boolean checks be a performance concern?

Reply:
Will have to look into performance.  I'll add a note about that in the description.  For now, we're under pressure to get something working.

Reply:
Adding a comment about that would help folks understand why it's if-def'd out

Reply:
New issue https://github.com/LLNL/axom/issues/1445

Reply:
See issue https://github.com/LLNL/axom/issues/1445